### PR TITLE
Support Recursive types

### PR DESCRIPTION
**Pull request by aggelgian**
This PR will allow CutEr to reason over recursive types.

Remaining tasks:
- [x] Support by the solver backend.
- [ ] Normalize type / spec definitions.
- [x] Add tests for recursive types.

### CODE REVIEW

**Comment by aggelgian** [comment]
Normalize type / spec definitions will be done in a separate PR.

File: src/cuter_types.erl [hunk 1]
```diff
@@ -966,3 +1055,19 @@ find_remote_deps(#t{kind = ?record_tag, rep = {Name, OverridenFields}}, Visited,
 %% all others
 find_remote_deps(_Type, _Visited, Deps, _StoredTypes) ->
   Deps.
+
+%% ----------------------------------------------------------------------------
+%% API for erl_type_defs().
+%% ----------------------------------------------------------------------------
+
+-spec erl_type_deps_map(fun((erl_type_dep()) -> any()), erl_type_deps()) -> [any()].
```
Comment (kostis) [inline src/cuter_types.erl]: this spec is more accurate as:
```erlang
-spec erl_type_deps_map(fun((erl_type_dep()) -> T), erl_type_deps()) -> [T].
````

File: src/cuter_pp.erl [hunk 2]
```diff
@@ -58,9 +60,11 @@
         , module_non_existing/1
         , mfa_non_existing/3
        ]).
-
+%% Various debug prints.
+-export([pp_typedeps/1, pp_type/1]).
```
Comment (kostis) [inline src/cuter_pp.erl]: No need to export these two functions.

File: src/cuter_pp.erl [hunk 3]
```diff
@@ -25,6 +25,8 @@
 %% Format errors.
 -export([abstract_code_missing/1, cover_compiled_module/1, non_existing_module/1,
          compilation_errors/2, non_existing_mfa/1]).
+%% Debug Verbose prints.
+-export([parsed_spec/1]).
 
 -export([ reversible_operations/1
```
Comment (kostis) [inline src/cuter_pp.erl]: Are these three exported functions used anywhere?

Comment (aggelgian) [inline src/cuter_pp.erl]: I think not currently, but let's clean them up in an another PR

**Pull request merged**